### PR TITLE
Add Django field generators.

### PR DESCRIPTION
**Pull request by adam-thomas**
@incuna/backend review please? :)

### CODE REVIEW

**Comment by adam-thomas** [comment]
bump

**Comment by adam-thomas** [comment]
Moar bump

File: surveys/generators.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      instance.  (The instance is intended to be a SurveyField, which will specify
+      the field's behaviour.)
+    - generate_field - return an actual field instance.  The instance is given
+      the kwargs from get_field_kwargs() but those can be overridden or added to by
+      passing more to the generate_field() call.
+    """
+    required = True
+
+    def get_field_kwargs(self, instance):
+        return {'required': self.required}
```
Comment (maxpeterson) [inline surveys/generators.py]: Now this is a library whether a field is `required`  should come from the model.  Do you want to address this here or as part of another PR?

Comment (maxpeterson) [inline surveys/generators.py]: @adam-thomas can you create an issue to make `is_required` a field on the model.

**Comment by adam-thomas** [comment]
Thanks :)

File: surveys/generators.py (hunk 1 above)
Comment (adam-thomas) [inline surveys/generators.py]: On the `SurveyField` model you mean?

**Pull request merged**